xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java

```diff
@@ -109,7 +109,7 @@ public String escapeElementEntities(String text)
             String result = super.escapeElementEntities(text);
 
             // "\r" characters are automatically transformed in &#xD; but we want to keep the original \r there.
-            return cleanAmpersandEscape(result).replaceAll("&#xD;", "\r");
+            return result.replaceAll("&#xD;", "\r");
```

Comment:
This method is only called for unicode characters, or it's also called for other html entities? If it's only for unicode characters your change makes sense, else I'm not so sure that we want to remove the call to `cleanAmpersandEscape`.

Reply:
This is called for all text nodes that are part of the parsed, sanitized DOM. It does not matter if they contain unicode or not.

The `cleanAmpersandEscape` has been removed exactly because of XCOMMONS-2265 : it replaces e.g. `&amp;#xD;`  by `&#xD;` (which is then displayed by browsers as as `<CR>`-control character), making it impossible for users to display a verbatim `&#xD;` in the display of the browser.  (Or to use any other encoded entity representation like "The Character α is represented by '&amp;#x3B1;'".  To display this information in HTML, one has to use:

    <p>The Character α is represented by &apos;&amp;#x3B1;&apos;</p>

(where the `&apos;` to represent the `'` is not mandatory, as we are outside an element tag, but the `&amp;` before the `#x3B1;` is, or the users would read "The Character α is represented by α", which is not the intended information.)

Reply:
@mflorea so my understanding is that `HTMLCleaner` behaviour did changed in 2.25 and it is actually main reason why @ClemensRobbenhaar wanted to upgrade, it fixes the bad escape of `&` see: https://sourceforge.net/p/htmlcleaner/bugs/221/